### PR TITLE
Add `torch-base`

### PR DESCRIPTION
**Pull request by wbrown**
Add the `torch-base` image which builds a `pytorch` image that doesn't crash on AMD EPYCs while finetuning.

### CODE REVIEW

**Review by brandonrjacobs — APPROVED** [review]

File: images/torch-base/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+#Install core packages
+RUN apt-get update && apt-get install -y \
+      libncurses5 python3 python3-pip curl git apt-utils ssh ca-certificates \
+      tmux nano vim sudo bash rsync htop wget unzip tini && \
+    update-alternatives --install /usr/bin/python python /usr/bin/python3 1 && \
+    update-alternatives --install /usr/bin/pip pip /usr/bin/pip3 1 && \
+    pip3 install --no-cache-dir --upgrade pip && \
+    apt-get clean
+RUN apt-get install -y libcurand-11-3 libcufft-11-3 libcublas-11-3 \
+         libnvtoolsext1 cuda-nvrtc-11-3 libcusparse-11-3 libcusolver-11-3
```
Comment (todie) [inline images/torch-base/Dockerfile:55]: woops

Comment (todie) [inline images/torch-base/Dockerfile]: missed an escaped carriage return here

**Review by todie — APPROVED** [review]

File: images/torch-base/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,62 @@
+## Build pytorch
+FROM nvidia/cuda:11.3.1-devel-ubuntu20.04 as builder
```
Comment (salanki) [inline images/torch-base/Dockerfile:2]: All our current nodes support cuda 11.4, any reason to not build using that?

File: images/torch-base/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      libncurses5 python3 python3-pip git apt-utils ssh ca-certificates \
+      python3-distutils python3-numpy build-essential cmake && \
+    update-alternatives --install /usr/bin/python python /usr/bin/python3 1 && \
+    update-alternatives --install /usr/bin/pip pip /usr/bin/pip3 1 && \
+    pip3 install --no-cache-dir --upgrade pip && \
+    apt-get clean
+
+RUN mkdir /build
+WORKDIR /build
+RUN git clone --recursive https://github.com/pytorch/pytorch -b v1.10.1 && \
```
Comment (salanki) [inline images/torch-base/Dockerfile:14]: Reason for not going with 1.11?

**Comment by salanki** [comment]
Except for image size, I'd love to know the difference between this and https://catalog.ngc.nvidia.com/orgs/nvidia/containers/pytorch, and some performance benchmarks.

File: images/torch-base/Dockerfile (hunk 2 above)
Comment (wbrown) [reply images/torch-base/Dockerfile:2]: We can try cuda 11.4 and see if that works.

File: images/torch-base/Dockerfile (hunk 3 above)
Comment (wbrown) [reply images/torch-base/Dockerfile:14]: It's what's tested and known -- I just looked at 1.11, and there's 3,300 commits in between 1.10.1 and 1.11. I would do an 1.11 only if we test with our inference stack, and that's probably a separate effort.

Comment (salanki) [reply images/torch-base/Dockerfile:14]: makes sense, thank you

**Comment by wbrown** [comment]
@salanki -- the main thing is which BLAS library is being used, as the Intel library crashes on AMD nodes specifically while I was training.

There's been no attempt with this image to optimize pytorch for AMD. That's something for later, I think.

**Comment by salanki** [comment]
I'd still like benchmark between NVIDIAs and this one just to know, if possible. Should be easy now that you have workflows!

**Comment by salanki** [comment]
I'm happy to merge this if we need it out for the finetune workflow, but I'd still like benchmarks.

File: images/torch-base/Dockerfile (hunk 2 above)
Comment (todie) [reply images/torch-base/Dockerfile:2]:
```yaml
# syntax=docker/dockerfile:1.2

ARG CUDA_VERSION=11.0.3
ARG RUNTIME=runtime
ARG DISTRO=ubuntu20.04

FROM nvidia/cudagl:${CUDA_VERSION}-${RUNTIME}-${DISTRO}
```

I do this

**Comment by todie** [comment]
Moved to coreweave/pytorch

**Pull request closed without merging**